modules/xfeatures2d/src/beblid.cpp

```diff
@@ -0,0 +1,602 @@
+                      {25, 29, 23, 26, 1, 0}, {15, 19, 15, 18, 1, 0}, {13, 22, 12, 25, 1, 0}, {1, 22, 1, 11, 1, 0},
+                      {14, 12, 14, 9, 1, 1}, {10, 27, 9, 23, 1, 2}, {9, 4, 6, 1, 1, 1}, {22, 12, 21, 16, 1, 0},
+                      {5, 27, 1, 28, 1, 1}, {30, 14, 28, 7, 1, 0}, {17, 9, 16, 21, 1, 2}, {17, 9, 17, 6, 1, 0},
+                      {4, 4, 1, 1, 1, 1}, {30, 2, 28, 5, 1, 0}, {18, 4, 17, 7, 1, 1}, {15, 13, 15, 10, 1, 1},
+                      {12, 30, 11, 26, 1, 2}, {16, 28, 15, 29, 1, 1}, {30, 11, 28, 11, 1, 0}, {9, 12, 8, 10, 1, 1},
+                      {22, 19, 21, 16, 1, 0}, {30, 20, 29, 26, 1, 0}, {22, 10, 20, 7, 1, 2}, {2, 2, 1, 5, 1, 0},
+                      {9, 9, 7, 9, 1, 0}, {27, 1, 25, 3, 1, 0}, {21, 23, 20, 25, 1, 1}, {10, 3, 8, 5, 1, 1},
+                      {24, 1, 23, 3, 1, 0}, {5, 29, 4, 28, 1, 0}, {27, 23, 26, 18, 1, 1}, {22, 2, 22, 1, 1, 0},
+                      {7, 20, 6, 19, 1, 0}, {12, 26, 9, 25, 1, 2}, {7, 1, 5, 2, 1, 0}, {2, 21, 1, 18, 1, 0},
+                      {2, 24, 1, 21, 1, 0}, {8, 17, 8, 14, 1, 0}, {30, 1, 28, 2, 1, 0}, {15, 30, 15, 28, 1, 0},
+                      {2, 5, 1, 9, 1, 0}, {18, 28, 17, 26, 1, 1}, {7, 29, 1, 30, 1, 1}, {17, 2, 17, 1, 1, 0},
+                      {21, 13, 21, 9, 1, 1}, {29, 15, 27, 15, 1, 0}, {28, 8, 27, 7, 2, 0}, {29, 14, 28, 18, 1, 0},
+                      {2, 26, 1, 30, 1, 1}, {16, 8, 16, 6, 1, 0}, {30, 26, 26, 24, 1, 0}, {15, 17, 15, 16, 6, 0},
+                      {30, 29, 27, 30, 1, 0}, {3, 30, 1, 28, 1, 0}, {17, 1, 16, 2, 1, 1}, {14, 30, 12, 30, 1, 1},
+                      {12, 17, 12, 16, 1, 0}, {4, 18, 4, 16, 1, 0}, {11, 4, 11, 1, 1, 1}, {21, 2, 18, 1, 1, 2},
+                      {16, 17, 16, 15, 5, 0}, {3, 1, 2, 2, 1, 0}, {23, 17, 23, 16, 1, 0}, {18, 12, 18, 11, 1, 0},
+                      {10, 28, 8, 30, 1, 0}, {12, 10, 12, 8, 1, 1}, {2, 14, 1, 9, 1, 1}, {6, 25, 6, 21, 1, 1},
+                      {6, 2, 2, 1, 1, 1}, {30, 19, 29, 20, 1, 0}, {25, 21, 23, 20, 1, 0}, {16, 10, 16, 9, 1, 0},};
+    }
+    else
+    {
+        CV_Error(Error::StsBadArg, "n_wls should be either SIZE_512_BITS or SIZE_256_BITS");
+    }
+}
+
+// Internal function that implements the core of BEBLID descriptor
+void BEBLID_Impl::computeBEBLID(const cv::Mat &integralImg,
+                                const std::vector<cv::KeyPoint> &keypoints,
+                                cv::Mat &descriptors)
+{
+    CV_DbgAssert(!integralImg.empty());
+    CV_DbgAssert(size_t(descriptors.rows) == keypoints.size());
+    const int *integralPtr = integralImg.ptr<int>();
+    cv::Size frameSize(integralImg.cols - 1, integralImg.rows - 1);
+
+    // Parallel Loop to process descriptors
+#ifndef CV_BEBLID_PARALLEL
+    const cv::Range range(0, keypoints.size());
+#else
+    cv::parallel_for_(cv::Range(0, int(keypoints.size())), [&](const Range &range)
+    {
+#endif
```

Comment:
BTW, You may swap `{` and `#endif` lines here and use this below

```
    }
#ifdef CV_BEBLID_PARALLEL
    );
#endif
```

Reply:
Good point! It's done.